OrbitGl/FunctionsDataView.cpp

```diff
@@ -156,6 +178,27 @@ void FunctionsDataView::OnContextMenu(const std::string& action, int menu_index,
   } else if (action == kMenuActionUnselect) {
     for (int i : item_indices) {
       GOrbitApp->DeselectFunction(*GetFunction(i));
+      // If a function is deselected, we have to make sure that the frame track is
+      // not created for this function on the next capture. However, we do not
+      // want to remove the frame track from the capture data.
+      GOrbitApp->DisableFrameTrack(*GetFunction(i));
+    }
+  } else if (action == kMenuActionEnableFrameTrack) {
+    for (int i : item_indices) {
+      const FunctionInfo& function = *GetFunction(i);
+      // Functions used as frame tracks must be hooked (selected), otherwise the
+      // data to produce the frame track will not be captured.
+      GOrbitApp->SelectFunction(function);
+      GOrbitApp->EnableFrameTrack(function);
+      GOrbitApp->AddFrameTrack(function);
+    }
+  } else if (action == kMenuActionDisableFrameTrack) {
+    for (int i : item_indices) {
+      // When we remove a frame track, we do not unhook (deselect) the function as
+      // it may have been selected manually (not as part of adding a frame track).
+      // However, disable the frame track, so it is not recreated on the next capture.
+      GOrbitApp->DisableFrameTrack(*GetFunction(i));
```

Comment:
What you describe (keep frame track hooked functions separate) sounds like quite a large change. 

I think it's fairly visible to users due to the icons (checkmark and "F") that a function gets hooked when selecting a frame track, so I'm not too concerned about this.

Reply:
I like the idea that Frametracks are completely independent from hooked functions, because I think it would make it more intuitive. The user does not need to know how frametracks work under the hood. 

But I also agree with Ronald, that it is completely fine how it is right now. Changing that would require a lot of work with rather little benefit.

Reply:
Agree. Just wanted to hear your ideas about that. That might be something for a future fixit week?
For now it should be fine.

Reply:
Sounds good, will resolve this for now.